llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -10304,8 +10304,11 @@ SDValue RISCVTargetLowering::lowerINSERT_SUBVECTOR(SDValue Op,
         return Op;
       }
 
+      // Use a insert_subvector that will resolve to an insert subreg.
+      unsigned Vscale = *VLen / RISCV::RVVBitsPerBlock;
```

Comment:
Do you think we should assert that `VLen` isn't zero? Some static analyzers might not like seeing a dangerous division. Given that we assert `VLen` earlier when `SubVecVT.isFixedLengthVector()` then perhaps it can work it out. Still, might be best to be explicit.